sound/soc/sof/sof-acpi-dev.c

```diff
@@ -231,8 +231,10 @@ static int sof_acpi_probe(struct platform_device *pdev)
 	}
 #endif
 
-	mach->mach_params.platform = dev_name(dev);
-	mach->mach_params.acpi_ipc_irq_index = desc->irqindex_host_ipc;
+	if (mach) {
+		mach->mach_params.platform = dev_name(dev);
+		mach->mach_params.acpi_ipc_irq_index = desc->irqindex_host_ipc;
+	}
 
 	sof_pdata->machine = mach;
```

Comment:
@bardliao should this be included in the check above? no point in assigning if mach is null I suppose

Reply:
@ranj063 It's harmless to assign a null to it and I think it will be more consistent for reading the code.